react-app-drawer-component/my-app/src/AppDrawer.js

```diff
@@ -0,0 +1,40 @@
+
+import { FaHamburger } from 'react-icons/fa';
+import './AppDrawer.css';
+
+/**
+ * AppDrawer is a hamburger menu component that toggles open and closed,
+ * and contains any number of pages.
+ * @param {heading} Represents name of menu.
+ * @param {items} An array containing pages listed within the menu.
+ */
+
+export default function AppDrawer({heading, items, onTitleChange, onMenuToggle, view}) {
+
+  return (
+    <div style={{position: 'absolute'}}>
+      {(view === 'main') && <FaHamburger className="hamburger-icon" onClick={onMenuToggle}/>}
```

Comment:
I see you put the "open/closed" state in the App, represented by `view === 'main'`. It's also fine for this state to be kept in the AppDrawer. That requires the overlay to be managed by AppDrawer, which is pretty reasonable.

If you do it that way, the AppDrawer knows whether or not it is open, and if it is, it displays the overlay.

The way you have it, the App needs to understand that when the menu is open it needs to display the overlay. That's information that seems would not normally need to be managed by the App.